op-node/rollup/derive/engine_update.go

```diff
@@ -117,15 +118,35 @@ func startPayload(ctx context.Context, eng ExecEngine, fc eth.ForkchoiceState, a
 // confirmPayload ends an execution payload building process in the provided Engine, and persists the payload as the canonical head.
 // If updateSafe is true, then the payload will also be recognized as safe-head at the same time.
 // The severity of the error is distinguished to determine whether the payload was valid and can become canonical.
-func confirmPayload(ctx context.Context, log log.Logger, eng ExecEngine, fc eth.ForkchoiceState, id eth.PayloadID, updateSafe bool) (out *eth.ExecutionPayload, errTyp BlockInsertionErrType, err error) {
-	payload, err := eng.GetPayload(ctx, id)
+func confirmPayload(ctx context.Context, log log.Logger, eng ExecEngine, fc eth.ForkchoiceState, id eth.PayloadID, updateSafe bool, agossip *async.AsyncGossiper) (out *eth.ExecutionPayload, errTyp BlockInsertionErrType, err error) {
+	var payload *eth.ExecutionPayload
+	// if the payload is available from the async gossiper, it means it was not yet imported, so we reuse it
+	var cachedPayload *eth.ExecutionPayload
+	if agossip != nil {
+		cachedPayload = agossip.Get()
+	}
+	if cachedPayload != nil {
+		payload = agossip.Get()
```

Comment:
```suggestion
		payload = cachedPayload
```

Since we only want to call `Get()` once to ensure we have a consistent view of the result.

Reply:
Yep, I think that's what I had intended, but obv did not end up that way. TY